frontend_tests/node_tests/compose.js

```diff
@@ -279,32 +279,61 @@ people.add(bob);
     var range_length = 0;
     var compose_value = $("#compose_textarea").val();
     var selected_word = "";
+    var cursor_pos = 0;
 
     global.document.queryCommandEnabled = function () {
         return queryCommandEnabled;
     };
     global.document.execCommand = function (cmd, bool, markdown) {
         var compose_textarea = $("#compose-textarea");
         var value = compose_textarea.val();
-        $("#compose-textarea").val(value.substring(0, compose_textarea.range().start)+
-            markdown+value.substring(compose_textarea.range().end, value.length));
+        $("#compose-textarea").val(value.substring(0, compose_textarea.range().start) +
+            markdown + value.substring(compose_textarea.range().end, value.length));
     };
+    function set_markdown_text_range(cursor_position, compose_msg) {
+        // It's the same logic which is used in frontend to extract the word
+        // present at cursor position for markdown.
+        var start_pos = cursor_position - 1;
+        var end_pos = cursor_position;
+        var next_char = compose_msg.charAt(start_pos);
+
+        while ((next_char !== " " && next_char !== "\n") && start_pos >= 0) {
+            start_pos -= 1;
+            next_char = compose_msg.charAt(start_pos);
+        }
+        next_char = compose_msg.charAt(end_pos);
+        while ((next_char !== " " && next_char !== "\n") && end_pos <= compose_msg.length) {
+            end_pos += 1;
+            next_char = compose_msg.charAt(end_pos);
+        }
+        return [start_pos + 1, end_pos - start_pos - 1];
+    }
 
     $("#compose-textarea").range = function () {
+        if (range_length === 0) {
+            // If text for markdown is not selected by user, then select the word present
+            // at cursor position.
+            var temp = set_markdown_text_range(cursor_pos, $("#compose-textarea").val());
+            range_start = temp[0];
+            range_length = temp[1];
+        }
+        // If there is already selected text, we don't need to set the range.
         return {
             start: range_start,
             end: range_start + range_length,
             length: range_length,
             range: noop,
-            text: $("#compose-textarea").val().substring(range_start, range_length+range_start),
+            text: $("#compose-textarea").val().substring(range_start, range_length + range_start),
         };
     };
     $('#compose-textarea').caret = noop;
 
     // Test bold: ctrl/cmd + b.
+
```

Comment:
Can you split out a separate commit for doing the reformatting in this file?  Some of these reformats, like this one, don't make sense to me.

Reply:
Removed. Don't require after rebase.